### PR TITLE
docs: ✏️ Changes the docs for basic architecture for plugin

### PR DESCRIPTION
**Pull request by mohitm15**
The doc has been revised as per the latest version regarding the basic
plugin architecture with reference.

✅ Closes: #1278

**What kind of change does this PR introduce?**

**Did you add tests for your changes?**
Not required 

**If relevant, did you update the documentation?**
Yes

**Summary**
The doc has been revised as per the latest version regarding the basic
plugin architecture with reference

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by ematipico** [comment]
Could you update the tests too? As you can see the CI is failing

File: packages/generate-plugin/templates/src/_index.js.tpl [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
  * See the webpack docs for more information about plugins:
- * https://github.com/webpack/docs/wiki/how-to-write-a-plugin
+ * https://webpack.js.org/contribute/writing-a-plugin/#basic-plugin-architecture
  */
 
-function <%= name %>(options) {
-	// Setup the plugin instance with options...
+class <%= name %> {
+  apply(compiler) {
+    compiler.hooks.done.tap('Hello World Plugin', (
```
Comment (anshumanv) [inline packages/generate-plugin/templates/src/_index.js.tpl]: Won't this make more sense?

```suggestion
    compiler.hooks.done.tap(<%= name %> (
```

File: packages/generate-plugin/templates/src/_index.js.tpl [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+      stats /* stats is passed as argument when done hook is tapped.  */
```
Comment (jamesgeorge007) [inline packages/generate-plugin/templates/src/_index.js.tpl]:
```suggestion
      stats /* stats is passed as an argument when done hook is tapped.  */
```

Comment (mohitm15) [reply packages/generate-plugin/templates/src/_index.js.tpl]: @jamesgeorge007 take reference from documentation

Comment (jamesgeorge007) [reply packages/generate-plugin/templates/src/_index.js.tpl]: The former isn't grammatically valid.

**Comment by snitin315** [comment]
@evilebottnawi This is fixed can be closed.

**Comment by anshumanv** [comment]
@mohitm15 are you working on this?

**Comment by evilebottnawi** [comment]
@snitin315 still not fixed, somebody should send a new PR

**Comment by anshumanv** [comment]
On it.

**Comment by anshumanv** [comment]
Closing in favour of https://github.com/webpack/webpack-cli/pull/1354

**Pull request closed without merging**